streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -400,36 +401,72 @@ public synchronized void onChange(final Thread thread,
     }
 
     /**
+     * @deprecated use {@link #KafkaStreams(Topology, Properties)} instead
+     */
+    @Deprecated
+    public KafkaStreams(final TopologyBuilder builder,
+                        final Properties props) {
+        this(builder.internalTopologyBuilder, new StreamsConfig(props), new DefaultKafkaClientSupplier());
+    }
+
+    /**
+     * @deprecated use {@link #KafkaStreams(Topology, StreamsConfig)} instead
+     */
+    @Deprecated
+    public KafkaStreams(final TopologyBuilder builder,
+                        final StreamsConfig config) {
+        this(builder.internalTopologyBuilder, config, new DefaultKafkaClientSupplier());
+    }
+
+    /**
+     * @deprecated use {@link #KafkaStreams(Topology, StreamsConfig, KafkaClientSupplier)} instead
+     */
+    @Deprecated
+    public KafkaStreams(final TopologyBuilder builder,
```

Comment:
I will clean up the deprecation warning in the last PR. Right now we have many that will resolve automatically when deprecating `KStreamBuilder`. Afterwards, I have a better picture, which one to suppress and which one not because they indicate required refactoring work...